### PR TITLE
feat(k8s): add warning if cluster type is not prefixed by kapsule or multicloud

### PR DESCRIPTION
**Pull request by pablo-ruth**
This PR add a warning if cluster type is not prefixed by kapsule or multicloud, instead of returning an error.

### CODE REVIEW

File: scaleway/resource_k8s_cluster.go [hunk 1]
```diff
@@ -335,6 +335,14 @@ func resourceScalewayK8SClusterCreate(ctx context.Context, d *schema.ResourceDat
 		clusterType = ""
 	}
 
+	if !slices.ContainsFunc([]string{"kapsule", "multicloud"}, func(s string) bool { return strings.HasPrefix(clusterType.(string), s) }) {
+		diags = append(diags, diag.Diagnostic{
+			Severity: diag.Warning,
+			Summary:  "Unexpected cluster type",
+			Detail:   fmt.Sprintf("The expected cluster type is one of %v, but got %s", []string{"kapsule", "multicloud", "kapsule-dedicated-*", "multicloud-dedicated-*"}, clusterType.(string)),
+		})
```
Comment (Codelax) [inline scaleway/resource_k8s_cluster.go:343]: You can add attribute path to have a better output:
```
AttributePath: cty.GetAttrPath("type"),
```

File: scaleway/resource_k8s_cluster.go [hunk 2]
```diff
@@ -3,6 +3,7 @@ package scaleway
 import (
 	"context"
 	"fmt"
+	"slices"
```
Comment (Codelax) [inline scaleway/resource_k8s_cluster.go:6]: We should bump to go 1.21

File: scaleway/resource_k8s_cluster.go (hunk 1 above)
Comment (pablo-ruth) [reply scaleway/resource_k8s_cluster.go]: done :)

File: scaleway/resource_k8s_cluster.go (hunk 2 above)
Comment (pablo-ruth) [reply scaleway/resource_k8s_cluster.go]: ok I replaced the call to slices by a simple if as there is only two elements

**Review by Codelax — APPROVED** [review]

**Pull request merged**